### PR TITLE
Handle conflict response in VBase client

### PR DESCRIPTION
**Pull request by grovina**
A plugin checks for conflict response code (409). In the positive case, it asks conflict resolver to act then retries.

Also removed unused `SetBucketState`.

### CODE REVIEW

File: vbase/client.go [hunk 1]
```diff
@@ -282,7 +276,39 @@ func (cl *client) DeleteFile(bucket, path string) error {
...
+
+func conflictHandler(cl *client, bucket string) plugin.Plugin {
+	p := plugin.New()
+	p.SetHandlers(plugin.Handlers{
+		"before dial": func(c *context.Context, h context.Handler) {
+			c.Request.Header.Set("X-Vtex-Detect-Conflicts", "true")
+			h.Next(c)
+		},
+		"response": func(c *context.Context, h context.Handler) {
+			if cl.conflictResolver != nil {
```
Comment (victorges) [inline vbase/client.go]: How about not even setting the handlers if `cl.conflictResolver` is `nil`?

File: vbase/client.go [hunk 2]
```diff
@@ -273,7 +267,7 @@ func (cl *client) ResolveConflicts(bucket string, patch PatchRequest) error {
 func (cl *client) DeleteFile(bucket, path string) error {
 	_, err := cl.http.Delete().
 		AddPath(fmt.Sprintf(pathToFile, cl.appName, bucket, path)).
-		Send()
+		Use(conflictHandler(cl, bucket)).Send()
```
Comment (victorges) [inline vbase/client.go]: **nit**: leave each chain call in its separate line? 🤔

File: vbase/client.go [hunk 3]
```diff
@@ -282,7 +276,39 @@ func (cl *client) DeleteFile(bucket, path string) error {
...
+					resolved, err := cl.conflictResolver.Resolve(cl, bucket)
+					if err != nil {
+						h.Error(c, fmt.Errorf("Error resolving conflicts in bucket %s: %v", bucket, err))
+					} else if !resolved {
+						h.Error(c, fmt.Errorf("Conflicts could not be solved in bucket %s", bucket))
+					}
+				}
+
+				// Retry
+				res, err := c.Client.Do(c.Request)
```
Comment (victorges) [inline vbase/client.go]: Does this work? I remember that I specifically had to clone the request before sending or it wouldn't work for a retry 🤔

Comment (victorges) [reply vbase/client.go]: Might this cause an infinite loop if the `conflictResolver` doesn't manage to actually resolve the conflicts?

File: vbase/responses.go [hunk 4]
```diff
@@ -26,15 +26,16 @@ type ConflictListResponse struct {
...
 	Mine   *ConflictEntry `json:"mine"`
+	Master *ConflictEntry `json:"master"`
 }
 
-// ConflictEntry is a Conflict's item
+// ConflictEntry holds a version (base, mine or master) of the conflicted file
 type ConflictEntry struct {
-	Value json.RawMessage `json:"value"`
+	Deleted bool            `json:"deleted"`
+	Value   json.RawMessage `json:"value"`
```
Comment (victorges) [inline vbase/responses.go]: Reminder to add the `MimeType` here

**Comment by grovina** [comment]
@victorges 🏓

File: vbase/responses.go [hunk 5]
```diff
@@ -36,6 +36,7 @@ type Conflict struct {
 
 // ConflictEntry holds a version (base, mine or master) of the conflicted file
 type ConflictEntry struct {
-	Deleted bool            `json:"deleted"`
-	Value   json.RawMessage `json:"value"`
+	Deleted  bool            `json:"deleted"`
+	MimeType string          `json:"mimeType"`
```
Comment (victorges) [inline vbase/responses.go]: **nit**: Btw MIME is actually an acronym so this should be all upper case per go's style guideline 🤔

**Pull request merged**